python/plugins/processing/ui/resultsdockbase.ui

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
+ <class>DockWidget</class>
+ <widget class="QDockWidget" name="DockWidget">
```

Comment:
@nyalldawson I implemented all suggestions except using `QgsDockWidget`. After editing .ui file and rebuilding, I get following error
```
Traceback (most recent call last):
  File "/home/alex/devel/qgis/build/output/python/qgis/utils.py", line 338, in startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "/home/alex/devel/qgis/build/output/python/plugins/processing/__init__.py", line 36, in classFactory
    from processing.ProcessingPlugin import ProcessingPlugin
  File "/home/alex/devel/qgis/build/output/python/qgis/utils.py", line 647, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/home/alex/devel/qgis/build/output/python/plugins/processing/ProcessingPlugin.py", line 43, in 
    from processing.gui.ResultsDock import ResultsDock
  File "/home/alex/devel/qgis/build/output/python/qgis/utils.py", line 647, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/home/alex/devel/qgis/build/output/python/plugins/processing/gui/ResultsDock.py", line 40, in 
    os.path.join(pluginPath, 'ui', 'resultsdockbase.ui'))
  File "/usr/lib/python3/dist-packages/PyQt5/uic/__init__.py", line 199, in loadUiType
    return (ui_globals[winfo["uiclass"]], getattr(QtWidgets, winfo["baseclass"]))
AttributeError: module 'PyQt5.QtWidgets' has no attribute 'QgsDockWidget'
```